engine/crates/engine-v2/src/sources/graphql/request/prepare.rs

```diff
@@ -196,23 +258,93 @@ impl QueryBuilderContext {
             .fields_ordered_by_parent_entity_id_then_position()
             .into_iter()
             .chunk_by(|field| field.definition().parent_entity_id);
+
         for (entity_id, fields) in entity_to_fields.into_iter() {
-            if maybe_entity_id != Some(entity_id) {
-                indent_write!(
-                    buffer,
-                    "... on {} {{\n",
-                    selection_set.walker().schema().walk(entity_id).name()
-                )?;
-                buffer.indent += 1;
-            }
-            for field in fields {
-                self.write_field(buffer, field)?;
-            }
-            if maybe_entity_id != Some(entity_id) {
-                buffer.indent -= 1;
-                indent_write!(buffer, "}}\n")?;
+            let fields = fields.collect_vec();
```

Comment:
Why allocate a vec? `write_entity_fields` can just accept an `impl Iterator<Item = PlanWalker<'_, FieldId>>`

Reply:
Ah got it